'moduleVersion' on null object

When running ./buckw targets, I get the following error message with okbuck 0.15.4: 

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:okbuck'.
> Cannot get property 'moduleVersion' on null object
```

Root build.gradle

```
// Top-level build file where you can add configuration options common to all sub-projects/modules.

buildscript {
  repositories {
    jcenter()
    maven { url 'https://maven.fabric.io/public' }
  }
  dependencies {

    classpath 'com.android.tools.build:gradle:2.2.2'
    classpath 'com.github.triplet.gradle:play-publisher:1.1.5'
    classpath 'com.neenbedankt.gradle.plugins:android-apt:1.8'
    classpath 'me.tatarka:gradle-retrolambda:3.4.0'
    classpath 'io.fabric.tools:gradle:1.22.0'
    classpath 'com.newrelic.agent.android:agent-gradle-plugin:5.8.4'
    classpath 'com.google.gms:google-services:3.0.0'
    classpath 'com.jakewharton:butterknife-gradle-plugin:8.4.0'
    classpath 'com.uber:okbuck:0.15.4'
  }


  allprojects {
    repositories {
      jcenter()
      flatDir { dirs 'libs' }
      maven { url "http://appboy.github.io/appboy-android-sdk/sdk" }
      maven { url 'https://maven.fabric.io/public' }
      maven { url 'https://clojars.org/repo/' }
      maven { url 'https://jitpack.io' }
    }
  }
}

apply plugin: 'com.uber.okbuck'


ext {
  versionCode = 451
  versionName = '4.1.0'

  minSdkVersion = 15
  targetSdkVersion = 25
  compileSdkVersion = 25
  buildToolsVersion = '25.0.2'

  supportLibraryVersion = '25.1.0'
  playServicesVersion = '10.0.1'

  daggerVersion = '1.2.5'
  butterKnifeVersion = '8.4.0'

  mParticleVersion = '4.9.6'

  junitVersion = '4.12'
  mockitoVersion = '2.2.1'
  androidSupportTestVersion = '0.5'

  minifyEnabled = true
}
```

@kageiit 

The "moduleVersion" issue is not reproduced with 0.14.4. But I am getting a new one on retrolambda jar location.

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':setupOkbuck'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:84)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:55)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:62)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:51)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:236)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.execute(DefaultTaskGraphExecuter.java:228)
        at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:61)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:228)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:215)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:77)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:58)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:32)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:113)
        at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
        at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
        at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
        at org.gradle.initialization.DefaultGradleLauncher$3.execute(DefaultGradleLauncher.java:196)
        at org.gradle.initialization.DefaultGradleLauncher$3.execute(DefaultGradleLauncher.java:193)
        at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:56)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:193)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:119)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:102)
        at org.gradle.launcher.exec.GradleBuildController.run(GradleBuildController.java:71)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:75)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:49)
        at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:44)
        at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:29)
        at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:47)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
        at org.gradle.util.Swapper.swap(Swapper.java:38)
        at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
        at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
        at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
        at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
        at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
        at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: java.lang.IllegalStateException: /Users/yuliya/Desktop/aaa/bbb/.okbuck/cache/retrolambda/retrolambda-2.3.0.jar must be located inside /Users/yuliya/Desktop/aaa/bbb
        at com.uber.okbuck.core.util.FileUtil.getRelativePath(FileUtil.java:30)
        at com.uber.okbuck.core.util.FileUtil$getRelativePath$0.call(Unknown Source)
        at com.uber.okbuck.core.dependency.DependencyCache.get(DependencyCache.groovy:70)
        at com.uber.okbuck.core.dependency.DependencyCache$get$0.call(Unknown Source)
        at com.uber.okbuck.core.model.base.Scope$_getExternalDeps_closure1.doCall(Scope.groovy:51)
        at com.uber.okbuck.core.model.base.Scope.getExternalDeps(Scope.groovy:50)
        at com.uber.okbuck.core.util.RetrolambdaUtil.fetchRetrolambdaDeps(RetrolambdaUtil.java:56)
        at com.uber.okbuck.core.util.RetrolambdaUtil$fetchRetrolambdaDeps.call(Unknown Source)
        at com.uber.okbuck.OkBuckGradlePlugin$_apply_closure1$_closure7.doCall(OkBuckGradlePlugin.groovy:143)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:596)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:577)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:95)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:76)
        ... 70 more
```

--- a/app/src/main/AndroidManifest.xml
+++ b/app/src/main/AndroidManifest.xml
@@ -4,7 +4,7 @@
     <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
 
     <application
-        android:name="com.uber.okbuck.example.AppShell"
+        android:name=".AppShell"
         android:allowBackup="true"
         android:icon="@mipmap/ic_launcher"
         android:label="@string/app_name"

--- a/buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidAppTarget.groovy
+++ b/buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidAppTarget.groovy
@@ -82,25 +82,7 @@ class AndroidAppTarget extends AndroidLibTarget {
         manifestXml.@'android:versionCode' = String.valueOf(versionCode)
         manifestXml.@'android:versionName' = versionName
         manifestXml.application.@'android:debuggable' = String.valueOf(debuggable)
-
-        def sdkNode = {
-            'uses-sdk'('android:minSdkVersion': String.valueOf(minSdk),
-                    'android:targetSdkVersion': String.valueOf(targetSdk)) {}
-        }
-        if (manifestXml.'uses-sdk'.size() == 0) {
-            manifestXml.appendNode(sdkNode)
-        } else {
-            manifestXml.'uses-sdk'.replaceNode(sdkNode)
-        }
-
-        def builder = new StreamingMarkupBuilder()
-        builder.setUseDoubleQuotes(true)
-        return (builder.bind {
-            mkp.yield manifestXml
-        } as String)
-                .replaceAll('\\{http://schemas.android.com/apk/res/android\\}', 'android:')
-                .replaceAll('xmlns:android="http://schemas.android.com/apk/res/android"', '')
-                .replaceFirst('<manifest ', '<manifest xmlns:android="http://schemas.android.com/apk/res/android" ')
+        return super.processManifestXml(manifestXml)
     }
 
     ExoPackageScope getExopackage() {

--- a/buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidLibTarget.groovy
+++ b/buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidLibTarget.groovy
@@ -23,9 +23,4 @@ class AndroidLibTarget extends AndroidTarget {
     ManifestMerger2.MergeType getMergeType() {
         return ManifestMerger2.MergeType.LIBRARY
     }
-
-    @Override
-    String processManifestXml(GPathResult manifestXml) {
-        return null
-    }
 }

--- a/buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidTarget.groovy
+++ b/buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidTarget.groovy
@@ -15,6 +15,7 @@ import com.uber.okbuck.core.model.base.Scope
 import com.uber.okbuck.core.model.java.JavaLibTarget
 import com.uber.okbuck.core.util.FileUtil
 import groovy.util.slurpersupport.GPathResult
+import groovy.xml.StreamingMarkupBuilder
 import org.gradle.api.JavaVersion
 import org.gradle.api.Project
 import org.gradle.api.tasks.compile.JavaCompile
@@ -242,7 +243,26 @@ abstract class AndroidTarget extends JavaLibTarget {
         return manifestPath
     }
 
-    abstract String processManifestXml(GPathResult manifestXml)
+    String processManifestXml(GPathResult manifestXml) {
+        def sdkNode = {
+            'uses-sdk'('android:minSdkVersion': String.valueOf(minSdk),
+                    'android:targetSdkVersion': String.valueOf(targetSdk)) {}
+        }
+        if (manifestXml.'uses-sdk'.size() == 0) {
+            manifestXml.appendNode(sdkNode)
+        } else {
+            manifestXml.'uses-sdk'.replaceNode(sdkNode)
+        }
+
+        def builder = new StreamingMarkupBuilder()
+        builder.setUseDoubleQuotes(true)
+        return (builder.bind {
+            mkp.yield manifestXml
+        } as String)
+                .replaceAll('\\{http://schemas.android.com/apk/res/android\\}', 'android:')
+                .replaceAll('xmlns:android="http://schemas.android.com/apk/res/android"', '')
+                .replaceFirst('<manifest ', '<manifest xmlns:android="http://schemas.android.com/apk/res/android" ')
+    }
 
     private void ensureManifest() {
         Set<String> manifests = getAvailable(baseVariant.sourceSets.collect { SourceProvider provider ->
@@ -293,11 +313,7 @@ abstract class AndroidTarget extends JavaLibTarget {
         packageName = manifestXml.@package
 
         String processedManifest = processManifestXml(manifestXml)
-        if (processedManifest) {
-            mergedManifest.text = processedManifest
-        } else {
-            mergedManifest.text = originalManifest
-        }
+        mergedManifest.text = processedManifest
     }
 
     static List<String> getJavaCompilerOptions(BaseVariant baseVariant) {

--- a/buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/ExoPackageScope.groovy
+++ b/buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/ExoPackageScope.groovy
@@ -27,13 +27,11 @@ class ExoPackageScope extends Scope {
         GPathResult manifestXml = slurper.parse(project.file(manifest))
         try {
             appClass = manifestXml.application.@"android:name"
-            appClass = appClass.replaceAll('\\.', "/")
-        } catch (Exception ignored) {
-        }
+            appClass = appClass.replaceAll('\\.', "/").replaceAll('^/', '')
+        } catch (Exception ignored) {}
         if (appClass != null && !appClass.empty) {
             base.sources.each { String sourceDir ->
-                FileTree found = project.fileTree(dir: sourceDir, includes:
-                        ["${appClass}.java"])
+                FileTree found = project.fileTree(dir: sourceDir, includes: ["**/${appClass}.java"])
                 if (found.size() == 1) {
                     appClass = FileUtil.getRelativePath(project.projectDir, found[0])
                 }

--- a/buildSrc/src/main/groovy/com/uber/okbuck/core/model/base/Scope.groovy
+++ b/buildSrc/src/main/groovy/com/uber/okbuck/core/model/base/Scope.groovy
@@ -81,11 +81,13 @@ class Scope {
         // get all first level external dependencies
         validConfigurations.collect {
             it.resolvedConfiguration.firstLevelModuleDependencies.each { ResolvedDependency resolvedDependency ->
-                ResolvedArtifact artifact = resolvedDependency.moduleArtifacts[0]
-                VersionlessDependency dependency = new VersionlessDependency(artifact.moduleVersion.id)
+                if (!resolvedDependency.moduleArtifacts.empty) {
+                    ResolvedArtifact artifact = resolvedDependency.moduleArtifacts[0]
+                    VersionlessDependency dependency = new VersionlessDependency(artifact.moduleVersion.id)
 
-                if (!depCache.getProject(dependency)) {
-                    firstLevel.add(dependency)
+                    if (!depCache.getProject(dependency)) {
+                        firstLevel.add(dependency)
+                    }
                 }
             }
         }

